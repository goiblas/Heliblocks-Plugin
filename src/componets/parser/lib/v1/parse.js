import React from "react";
import HtmlToReact, { Parser } from "html-to-react";

const processNodeDefinitions = new HtmlToReact.ProcessNodeDefinitions(React);

const defaultProcess = {
  shouldProcessNode: function () {
    return true;
  },
  processNode: processNodeDefinitions.processDefaultNode,
};

function isValidNode() {
  return true;
}

const parser = new Parser();

/**
 * 
 * @param {Object} config - Object config parse
 * @param {String} config.html - Html string to parse
 * @param {Array} config.processes - Instructions to parse html
 * @param {Array} config.preprocessing - Instructions before parse
 */
export const parse = ({ html, processes, preprocessing }) => {
  return parser.parseWithInstructions(
    html,
    isValidNode,
    [...processes, defaultProcess],
    preprocessing
  );
};
