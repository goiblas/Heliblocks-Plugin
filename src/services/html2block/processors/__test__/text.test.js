import React from "react";
import { textProcessor } from "..";
import { render } from "@testing-library/react";

import parse from "../..";
import Store from "../../../store";

describe("Text", () => {
  test("should process text edit", () => {
    const store = new Store([], () => {});
    const Result = parse({
      html: "<p>Hello World!</p>",
      store,
      processors: [textProcessor]
    });
    const { container } = render(<Result.Edit />);
    expect(container).toMatchSnapshot();
  });
});
