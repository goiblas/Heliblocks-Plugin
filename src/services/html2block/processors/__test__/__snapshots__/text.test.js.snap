// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should process text edit 1`] = `
<div>
  <input
    type="text"
    value="Hello World!"
  />
</div>
`;
